server/src/test/java/org/elasticsearch/index/mapper/DateFieldTypeTests.java

```diff
@@ -142,6 +143,46 @@ private void doTestIsFieldWithinQuery(DateFieldType ft, DirectoryReader reader,
         assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03", false, false, zone, null, context));
         assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03", false, true, zone, null, context));
         assertEquals(Relation.INTERSECTS, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03", true, false, zone, null, context));
+        // Bad dates
+        assertThrows(
+            ElasticsearchParseException.class,
+            () -> ft.isFieldWithinQuery(reader, "2015-00-01", "2016-04-03", randomBoolean(), randomBoolean(), zone, null, context)
+        );
+        assertThrows(
+            ElasticsearchParseException.class,
+            () -> ft.isFieldWithinQuery(reader, "2015-01-01", "2016-04-00", randomBoolean(), randomBoolean(), zone, null, context)
+        );
+        assertThrows(
+            ElasticsearchParseException.class,
+            () -> ft.isFieldWithinQuery(reader, "2015-22-01", "2016-04-00", randomBoolean(), randomBoolean(), zone, null, context)
```

Comment:
I see tests for `0` month and `0` days. I'm curious what happens with `0` years?

Reply:
I think 0 years should be valid? I mean there's no actual year 0 in UTC/Gregorian, but Java treats year 0 as 1 BCE and it's a valid date I think.